### DISCUSSION TITLE
Consideration of the stiffnesses of the beam / column connections.

### DISCUSSION THREAD

**Discussion by sinamitnn** [FEALiTE/FEALiTE2D#16] [Ideas]
According to my knowledge, at the moment we can not consider the stiffnesses of the beams connection.
We can have a fixed connection or a pin connection, but it would be really great to consider the known stiffnesses of beam connections.

**Comment by SaMohamed** [comment]
Hi Sinan,
Indeed there is no direct way in the library for modeling partial fixity between a column and a beam. But, you can insert a spring element at both beam/column ends and reduce the stiffness of the spring in the required direction and assign a big number for stiffness in the fixed direction.

**Comment by sinamitnn** [comment]
Hello Mohammed, 
thanks again for your answer. 
With the first opportunity, i am going to try to make an example, and going to compare it with Rstab.
I am going to share the results and the example model here. 

Thanks. 

Regards,

S.

**Comment by sinamitnn** [comment]
Hi Muhammed, 

i tried to model a simple frame example in order to use the springs as springs, but unfortunatelly i could not achieve what i wanted to do.

you can find my example frame below:

![grafik](https://github.com/FEALiTE/FEALiTE2D/assets/147400466/9ff00db7-1b93-4b80-9944-a6268d464fc8)

All the profiles are IPE80, and i wanted to have a connection stiffness of 80 kNm/rad. 
I made the model with fiixed supports. 

My fealite model is as its given below:

***

FEALiTE2D.Structure.Structure structure = new FEALiTE2D.Structure.Structure();

            Node2D n1 = new Node2D(0, 0, "n1");
            Node2D n2 = new Node2D(0, 5, "n2");
            Node2D n3 = new Node2D(0.02, 5, "n3");
            Node2D n4 = new Node2D(4.98, 5, "n4");
            Node2D n5 = new Node2D(5, 5, "n5");
            Node2D n6 = new Node2D(5, 0, "n6");


            n1.Support = new NodalSupport(true, true, true);
            n6.Support = new NodalSupport(true, true, true);

            structure.AddNode(n1, n2, n3, n4, n5, n6);

            IMaterial material = new GenericIsotropicMaterial() { E = 210000000, U = 0.3, Label = "Steel", Alpha = 1, Gama = 1, MaterialType = MaterialType.Steel };
            IFrame2DSection section = new Generic2DSection(0.00076, 1, 1, 0.0000008, 1, 1, 1, 1, material); // IPE80 => used for the tests

            FrameElement2D e1 = new FrameElement2D(n1, n2, "e1") { CrossSection = section };

            //FrameElement2D e2 = new FrameElement2D(n2, n3, "e2") { CrossSection = section };
            SpringElement2D e2 = new SpringElement2D(n2, n3, "e2");
            e2.R = 80;
            e2.K = 1e9;
            

            FrameElement2D e3 = new FrameElement2D(n3, n4, "e3") { CrossSection = section };

            //FrameElement2D e4 = new FrameElement2D(n4, n5, "e4") { CrossSection = section };
            SpringElement2D e4 = new SpringElement2D(n4, n5, "e4");
            e4.R = 80;
            e4.K = 1e9;


            FrameElement2D e5 = new FrameElement2D(n5, n6, "e5") { CrossSection = section };
         
            structure.AddElement(new[] { e1, e3, e5 });
            structure.AddElement(new[] { e2, e4 });
           
            


            LoadCase loadCase = new LoadCase("live", LoadCaseType.Live);
            structure.LoadCasesToRun.Add(loadCase);

            n2.NodalLoads.Add(new NodalLoad(5,  0, 0.00000000, LoadDirection.Global, loadCase));

            structure.LinearMesher.NumberSegements = 10;
            structure.Solve();

            MessageBox.Show(structure.Results.GetSupportReaction(n1, loadCase).ToString());
            MessageBox.Show(structure.Results.GetSupportReaction(n6, loadCase).ToString());


***
 
The problem with the element spring is, i can not transfer the loads in y-direction. 
We can transfer longitudinal loads with K and moments with R, but i think i need one more spring parameter(?).

How would you model this frame with Fealite?


BR, 

Sinan.

**Comment by SaMohamed** [comment]
Hi Sinan.
Sorry for responding late. 
Refrain from this idea as the spring is one directional and will give you wrong results.
